sklearn/model_selection/_split.py

```diff
@@ -2236,10 +2237,150 @@ def _build_repr(self):
         finally:
             warnings.filters.pop(0)
         params[key] = value
-
     return '%s(%s)' % (class_name, _pprint(params, offset=len(class_name)))
 
 
+class GroupTimeSeriesSplit(_BaseKFold):
+    """Grouped Time Series cross-validator
+
+    Provides train/test indices to split time series data samples
+    that are observed at fixed time intervals according to a
+    third-party provided group.
+    In each split, test indices must be higher than before, and thus shuffling
+    in cross validator is inappropriate.
+
+    This cross-validation object is a variation of :class:`KFold`.
+    In the kth split, it returns first k folds as train set and the
+    (k+1)th fold as test set.
+
+    The same group will not appear in two different folds (the number of
+    distinct groups has to be at least equal to the number of folds).
+
+    Note that unlike standard cross-validation methods, successive
+    training sets are supersets of those that come before them.
+
+    Read more in the :ref:`User Guide <cross_validation>`.
+
+    Parameters
+    ----------
+    n_splits : int, default=5
+        Number of splits. Must be at least 2.
+
+    max_train_size : int, default=None
+        Maximum size for a single training set.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.model_selection import GroupTimeSeriesSplit
+    >>> groups = ['a' ,'a' ,'a' ,'b','b' , 'b','c' ,'c' ,'c' ,'d' ,'d' ,'d']
+    >>> n_samples = len(groups)
+    >>> X = y = np.ones(n_samples)
+    >>> gtscv = GroupTimeSeriesSplit(n_splits=2)
+    >>> for train_index, test_index in gtscv.split(X, y, groups):
+    ...     print("TRAIN:", train_index, "TEST:", test_index)
+    TRAIN: [0, 1, 2, 3, 4, 5] TEST: [6, 7, 8]
+    TRAIN: [0, 1, 2, 3, 4, 5, 6, 7, 8] TEST: [9, 10, 11]
+    """
```

Comment:
I would add here, besides the indices, also the groups (as we have done in the discussion), so that it is much clearer.

Reply:
Done .